src/gui/sdlmain.cpp

```diff
@@ -3057,20 +3059,27 @@ static void launchcaptures(std::string const& edit) {
 	}
 	Cross::CreatePlatformConfigDir(path);
 	path += file;
-	Cross::CreateDir(path);
-	struct stat cstat;
-	if(stat(path.c_str(),&cstat) || (cstat.st_mode & S_IFDIR) == 0) {
-		printf("%s doesn't exists or isn't a directory.\n",path.c_str());
-		exit(1);
+
+	if (create_dir(path.c_str(), 0700) != 0) {
+		bool dir_exists = false;
+		if (errno == EEXIST) {
+			struct stat cstat;
+			if ((stat(path.c_str(), &cstat) == 0) &&
+			    (cstat.st_mode & S_IFDIR))
+				dir_exists = true;
```

Comment:
It would be nice to encapsulate this logic in `create_dir`, to avoid duplicate these post-fail checks "do we actually have the directory or was it blocked because a filename exists in it's place?".

`create_dir` could take a boolean with a meaning like "pass if directory exists".

Reply:
Creating `dir_exists` would open the gate for a plethora of additional functions like `file_exists`, `link_exists`, etc - I don't want to do that.

Adding boolean parameters to functions is a [code smell](https://alexkondov.com/should-you-pass-boolean-to-functions/), so I preferred to avoid it; I also wanted to make the function a drop-in replacement.

Decided to add an optional `flags` parameter, with only a single flag supported for now (but I can easily imagine adding few more - e.g. option for recursive creation of directories).

Reply:
> Adding boolean parameters to functions is a code smell

I agree; and as usual your implementation is top-notch!  

If and when the need arises, having more flags available (and the ability to combine them elegantly) will produce very readable code.